R/epi.R

```diff
@@ -38,3 +31,51 @@ treatment_scaling <- function(x, treatment_scaler, baseline_treatment = 0){
     dplyr::select(-"ft")
   return(x)
 }
+
+#' Disability adjusted life years
+#'
+#' Calculate Years Lived with Disability (YLDs) and Disability-Adjusted Life-Years
+#' based on disability weights from the Global Burden of Disease study. To estimate
+#' YLL we assume the average life expectancy for a person aged x years taken from the UN WPP
+#' Africa profile for 2022. To estimate the expected age in a given raneg we assume
+#' exponentially distributed ages with the range.
+#'
+#' Disability weights sourced \href{https://ghdx.healthdata.org/record/ihme-data/gbd-2017-disability-weights}{here}
+#'
+#' This is an approximation of YLD estimation from the GBD study; disability due to comorbid conditions
+#' such as motor impairment and anemia are excluded.
+#'
+#' Note: weigths and lengths are currently for Plasmodium falciparum
+#'
+#' @inheritParams get_rates
+dalys <- function(
+    x,
+    age_divisor,
+    mild_disability_weight = 0.006,
+    moderate_disability_weight = 0.051,
+    severe_disability_weight = 0.133,
+    clinical_episode_length = 0.01375,
+    severe_episode_length = 0.04795
+)
+{
+  x |>
+    dplyr::mutate(
+      e_age = round(
+        expected_age(
+          lower_age = (.data$age_lower * age_divisor) / 365, # Note this is always in years for DALYs
+          upper_age = (.data$age_upper * age_divisor) / 365, # Note this is always in years for DALYs
+          average_age = life_expectancy_africa[life_expectancy_africa$age == 0, "life_expectancy"]
+        )
+      )
+    ) |>
+    dplyr::left_join(
+      life_expectancy_africa, by = c("e_age" = "age")
+    ) |>
+    dplyr::mutate(
+      case_disability_weight = ifelse(.data$e_age <= 5, moderate_disability_weight, mild_disability_weight),
+      yld_pp = .data$severe * severe_disability_weight * severe_episode_length +
```

Comment:
Since we incorporate a treatment scaler for mortality calculations, we may want to do the same for disability (under the assumption that cases that are treated are no longer disabling. Would likely have a small impact but could be done for consistency.